puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/packaging.groovy

```diff
@@ -0,0 +1,73 @@
+def packages_to_build
+
+pipeline {
+    agent { label 'rpmbuild' }
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        ansiColor('xterm')
+        buildDiscarder(logRotator(numToKeepStr: '15'))
+    }
+
+    stages {
+        stage('Clone Packaging') {
+            steps {
+
+                deleteDir()
+                checkout changelog: true, poll: false, scm: [
+                    $class: 'GitSCM',
+                    branches: [[name: '${ghprbActualCommit}']],
+                    doGenerateSubmoduleConfigurations: false,
+                    extensions: [[$class: 'PreBuildMerge', options: [fastForwardMode: 'FF', mergeRemote: 'origin', mergeStrategy: 'default', mergeTarget: '${ghprbTargetBranch}']]],
+                    submoduleCfg: [],
+                    userRemoteConfigs: [
+                        [refspec: '+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/*', url: 'https://github.com/${ghprbGhRepository}']
+                    ]
+                ]
+
+            }
+        }
+        stage('Find Packages to Build') {
+            steps {
+
+                script {
+                    changed_packages = sh(returnStdout: true, script: "git diff origin/${ghprbTargetBranch} --name-only -- 'packages/**.spec' | cut -d'/' -f2 |sort -u").trim()
+                    packages_to_build = changed_packages.split().join(' ')
+                    update_build_description_from_packages(packages_to_build)
+                }
+
+            }
+        }
+        stage('Scratch Build Packages') {
+            steps {
+
+                script {
+                    obal {
```

Comment:
This has to be in a `script` call even though its a function?

Reply:
This is not The Greatest Pipeline in the World, no.
This is just a tribute.

https://github.com/theforeman/foreman-infra/blob/4afb997715c82d9d76082f8c6567fe0b6f4a75cd/puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/tfmROR51.groovy#L45-L51

But I'll try without.

Reply:
+1 I think that was originally mis-guided and custom functions allow you to avoid the `script` wrapper.